### DISCUSSION TITLE
API to figure out whether there is any changes in the form?

### DISCUSSION THREAD

**Discussion by go-mad** [react-hook-form/react-hook-form#3465] [Q&A] [answered]
Is there any API to figure out whether the form has been updated from the initial values? e.g. if my form is initially:
```
{
 first_name: "test_first",
 last_name: "test_last"
}
```
and after updating it becomes:
```
{
 first_name: "test_first_updated",
 last_name: "test_last
}
```
Can I get to know the changes?

**Comment by bluebill1049** [comment, marked as answer]
dirtyFields & isDirty?

**Reply by go-mad** [reply]
Thanks for the reply. It's not working for me: https://codesandbox.io/s/nostalgic-sinoussi-vl824?file=/src/App.js

**Reply by bluebill1049** [reply]
you are missing `defaultValues` at the useForm

**Reply by go-mad** [reply]
Okay so basically `defaultValues` is mandatory for this to work?

**Reply by bluebill1049** [reply]
yes, to compare

**Reply by go-mad** [reply]
After providing `defaultValues` also it doesn't work. In the example given in the docs works even without `defaultValues`: https://codesandbox.io/s/react-hook-form-v6-setvalue-wjplb

**Reply by bluebill1049** [reply]
carefully review the following CSB: 

https://codesandbox.io/s/condescending-sara-fkr0f?file=/src/App.js

- defaultValues
- setValue API

**Reply by go-mad** [reply]
Ahh! my mistake wrong nested names and missing `shouldDirty`. Thanks @bluebill1049

**Reply by bluebill1049** [reply]
no worries at all! not a mistake, just learning on the way buddy.
